plugins/orchestrator-backend/src/types/pagination.ts

```diff
@@ -8,23 +10,33 @@ export interface Pagination {
 }
 
 export function buildPagination(req: Request): Pagination {
-  const pagination: Pagination = {};
+  const pagination: Pagination = {
+    limit: undefined,
+    offset: undefined,
+    order: undefined,
+    sortField: undefined,
+  };
 
-  if (!isNaN(Number(req.query.page))) {
-    pagination.offset = Number(req.query.page);
+  if (!req.body?.paginationInfo) {
+    return pagination;
   }
+  const { page, pageSize, orderBy, orderDirection } = req.body
+    .paginationInfo as PaginationInfoDTO;
 
-  if (!isNaN(Number(req.query.pageSize))) {
-    pagination.limit = Number(req.query.pageSize);
+  if (!isNaN(Number(page))) {
+    pagination.offset = Number(page);
```

Comment:
To fix an already existing bug,
This should be
pagination.offset = Number(page) * Number(pageSize)

Reply:
what if `pageSize` is not defined?